lib/Compilation.js

```diff
@@ -1064,10 +1075,12 @@ class Compilation extends Tapable {
 			context,
 			entry,
 			module => {
+				this.hooks.buildEntry.call(module);
 				this.entries.push(module);
 			},
 			(err, module) => {
 				if (err) {
+					this.hooks.failedEntry.call(module, err);
```

Comment:
pass `entry` and `name` as arguments. Remove the `module` argument (it's not set).
`entry, name, err`